ui/selectors/assets.ts

```diff
@@ -224,6 +233,54 @@ export const getMultiChainAssets = createDeepEqualSelector(
   },
 );
 
+/**
+ * Gets a {@link Token} (EVM or Multichain) owned by the passed account by address and chainId.
+ *
+ * @param state - Redux state object
+ * @param tokenAddress - Token address (Hex for EVM, or CaipAssetType for non-EVM)
+ * @param chainId - Chain ID (Hex for EVM, or CaipChainId for non-EVM)
+ * @param internalAccount - The account holding the token to search for
+ * @returns Token object
+ */
+export const getTokenByAccountAndAddressAndChainId = createDeepEqualSelector(
+  (state) => state,
+  (_state, account?: InternalAccount) => account,
+  (
+    _state,
+    _account?: InternalAccount,
+    tokenAddress?: Hex | CaipAssetType | string,
+  ) => tokenAddress,
+  (
+    _state,
+    _account?: InternalAccount,
+    _tokenAddress?: Hex | CaipAssetType | string,
+    _chainId?: Hex | CaipChainId,
+  ) => _chainId,
+  (
+    state,
+    account?: InternalAccount,
+    tokenAddress?: Hex | CaipAssetType | string,
+    chainId?: Hex | CaipChainId,
+  ) => {
```

Comment:
Can we ensure that the memoization here still works ? I'm worried that with the extensive operations the memoization will not work.